src/test/java/sysc4806/graduateAdmissions/model/TermTest.java

```diff
@@ -0,0 +1,120 @@
+package sysc4806.graduateAdmissions.model;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Tests the functionality of the Term class
+ *
+ * @author Eric
+ */
+public class TermTest {
+
+    private Term term;
+    private Long id = 1L;
+    private String deadline = "deadline";
+    private Season season = Season.WINTER;
+    private String year = "year";
+
+
+    @BeforeEach
+    void setUp() {
+        term = new Term();
+    }
+
+    @Test
+    /* Create a Term without arguments and verify correct defaults are set */
+    public void testNoArgsConstructor() {
+        assertNotNull(term);
+        assertNotNull(term.getId());
+        assertFalse(term.isActive());
+        assertNull(term.getDeadline());
+        assertNull(term.getSeason());
+        assertNull(term.getYear());
+    }
+
+    @Test
+    /* Create a Term with all arguments and verify correct defaults are set */
+    public void testAllArgsConstructor() {
+        term = new Term(deadline, season, year, true);
+
+        assertNotNull(term);
+        assertNotNull(term.getId());
+        assertTrue(term.isActive());
+        assertEquals(deadline, term.getDeadline());
+        assertEquals(season, term.getSeason());
+        assertEquals(year, term.getYear());
+    }
+
+    @Test
+    /* Test that setActive sets it active */
+    public void testSetActive() {
+        term.setActive(true);
+
+        assertNotNull(term.getId());
+        assertTrue(term.isActive());
+        assertNull(term.getDeadline());
+        assertNull(term.getSeason());
+        assertNull(term.getYear());
+    }
+
+    @Test
+    /* Test that setDeadline sets a deadline */
+    public void testSetDeadline() {
+        term.setDeadline(deadline);
+
+        assertNotNull(term.getId());
+        assertFalse(term.isActive());
+        assertEquals(deadline, term.getDeadline());
+        assertNull(term.getSeason());
+        assertNull(term.getYear());
+    }
+
+    @Test
+    /* Test that setSeason sets the season */
+    public void testSetSeason() {
+        term.setSeason(season);
+
+        assertNotNull(term.getId());
+        assertFalse(term.isActive());
+        assertNull(term.getDeadline());
+        assertEquals(season, term.getSeason());
+        assertNull(term.getYear());
+    }
+
+    @Test
+    /* Test that setYear sets the year */
+    public void testSetYear() {
+        term.setYear(year);
+
+        assertNotNull(term.getId());
+        assertFalse(term.isActive());
+        assertNull(term.getDeadline());
+        assertNull(term.getSeason());
+        assertEquals(year, term.getYear());
+    }
+
+    @Test
+    /* Test equals is true given two identical Terms */
+    public void testEquals() {
+        term = new Term(deadline, season, year, true);
+        term.setId(id);
```

Comment:
If you want to avoid doing things like this, where you use a constructor and then set more fields after, you could specify a builder annotation on the Term object. Not a required change, just a thought. you can see how builders work by looking at Privilege and PrivilegeTest

Reply:
It would be an easy change, so if there's another required change I'll make the fix